front-end/src/app/shared/components/inputs/amount-input/amount-input.component.ts

```diff
@@ -74,7 +74,7 @@ export class AmountInputComponent extends BaseInputComponent implements OnInit,
   }
 
   updateMemoItemWithDate(date: Date) {
-    if (this.report?.coverage_from_date && this.report?.coverage_through_date) {
+    if (date && this.report?.coverage_from_date && this.report?.coverage_through_date) {
```

Comment:
By adding this, the else below  doesn't get hit .  If we set a date outside and then erase it, this.dateIsOutsideReport stays true. So clicking on the memo box throws the alert

<img width="521" alt="image" src="https://github.com/fecgov/fecfile-web-app/assets/97105825/425ca714-95f0-4eab-834a-1e9bc398c2a1">